internal/webhook/templatemanagement_webhook.go

```diff
@@ -113,8 +126,8 @@ func getManagedClustersForTemplate(ctx context.Context, cl client.Client, namesp
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (v *TemplateManagementValidator) ValidateDelete(ctx context.Context, _ runtime.Object) (admission.Warnings, error) {
-	return nil, nil
+func (v *TemplateManagementValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+	return nil, ErrTemplateManagementDeletionForbidden
```

Comment:
I think we should add ownerreference to the management object instead. And reject TemplateManagement deletion only when Management deletion timestamp is unset

Reply:
I found it excessive since the `TemplateManagement` will be automatically removed after the `Management` object deletion once all the CRDs are removed. But in case we change the deletion logic sometimes your approach makes sense.

Reply:
Yes, I understand, but I think it's better not to rely on the side effect of CRD removal when we have an explicit dependency